pkg/ocicni/ocicni.go

```diff
@@ -27,6 +27,8 @@ type cniNetworkPlugin struct {
 	defaultNetName string
 	networks       map[string]*cniNetwork
 
+	podCidr string
+
```

Comment:
Shouldn't podCidr be specific to the network?   Maybe put this in `type cniNetwork struct`? 

How else do you know which network the podCidr is to be applied to?

Reply:
PodCidr is a runtime config passed from CRI, for example, if node in kubernetes has valid PodCIDR in its status, dockershim CRI will get it and pass it to cni manager.
It's not a config owned by a specific network but a config owned by cni manager.
When cni manager call cni plugin, it will build a runtime config which carries the runtime info (including PortMapping, Bandwidth, IpRanges and so on) in its [CapabilityArgs](https://github.com/containernetworking/cni/blob/ec5ec68fa4979e8f98c13845dbc2352380ad179b/libcni/api.go#L49), then a cni network with specific format will recognize and use it, here is the [function](https://github.com/containernetworking/cni/blob/ec5ec68fa4979e8f98c13845dbc2352380ad179b/libcni/api.go#L133).
About runtime configuration, here is the [official documentation.](https://github.com/containernetworking/cni/blob/master/CONVENTIONS.md)

Reply:
This is not accurate. CRI by definition does not support multiple networks and expects all network info (IP, interfaces, runtimeConfig, PodCIDR, etc) to apply to the default cluster network (eg either the explicitly specified default network or the first-file-in-/etc network).

Now if we're talking about podman specifically (or any other non-CRI user of ocicni) then yes we should be able to send different runtimeConfig to different networks, but podman currently has no way to attach specific info to each selected network. That would have to be fixed first.

The solution is not to send the same config to every network though because that is wrong in some cases.

Reply:
> This is not accurate. CRI by definition does not support multiple networks and expects all network info (IP, interfaces, runtimeConfig, PodCIDR, etc) to apply to the default cluster network (eg either the explicitly specified default network or the first-file-in-/etc network).

I agree with this, CRI should have ability to find out its default network, but this network may contain several network plugins.

> Now if we're talking about podman specifically (or any other non-CRI user of ocicni) then yes we should be able to send different runtimeConfig to different networks, but podman currently has no way to attach specific info to each selected network. That would have to be fixed first.

What do you mean about **different networks**, I think there are no different networks here, only different network plugins, and these network plugins are only for one pod.

> The solution is not to send the same config to every network though because that is wrong in some cases.

Just as above, I think i just send the same config from pod or something else to every network plugin of one single network but not every network. Can you show some wrong cases?

Reply:
Yeah, I think I see what you mean, we should consider both two scenarios about CRI and podman, so global podCidr may not work properly in podman scenario, what we need is a 
map like **map[network]podCidr**, right ?